google/cloud/sql/connector/connector.py

```diff
@@ -463,13 +468,15 @@ def close(self) -> None:
                 self._loop.call_soon_threadsafe(self._loop.stop)
             # wait for thread to finish closing (i.e. loop to stop)
             self._thread.join()
+        self._closed = True
```

Comment:
this is not necessary as synchronous close calls `close_async` so should get closed by `close_async`